pkg/asset/agent/installconfig.go

```diff
@@ -255,6 +255,31 @@ func (a *OptionalInstallConfig) GetBaremetalHosts() []*baremetal.Host {
 	return nil
 }
 
+// ValidateBMCConfiguration checks that the Provisioning fields are configured when baremetal hosts BMC is set.
+func (a *OptionalInstallConfig) ValidateBMCConfiguration() {
+	if a.Config == nil || a.Config.Platform.Name() != baremetal.Name {
+		return
+	}
+
+	for _, host := range a.GetBaremetalHosts() {
+		if host.BMC.Address != "" {
```

Comment:
the happy path here could help in reducing the nesting level

Reply:
Fixed.